### PR TITLE
created 3 classes

### PR DESCRIPTION
**Pull request by TuyishimireEric**
# OOP school library: add Person, Student and Teacher classes
In this project, I have started building a school library app. In this initial step, I implemented the classes to represent students and teachers.

## Highlights
- [x]  Created class Person with the given methods
- [x]  Created class Students with the given methods
- [x]  Created class Teacher with the given methods

### CHEERS AND HAPPY REVIEWING!! :joy_cat:

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..100)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? >= 18 or @parent_permission
```
Comment (FrederickMih) [inline person.rb]: - [x] Based on the [section requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md) of this project, the **and** and **or** keywords are banned. Always use **&&** and **||** instead.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Person
+  attr_accessor :id, :name, :age, :parent_permission
```
Comment (FrederickMih) [inline person.rb]: - [x] A person `id` should only be retrieved and should not be modified. Let's kindly fix this by using only Getter(`attr_reader`) for the `id` that's

`attr_accessor :id`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+# OOP school library: add Person, Student and Teacher classes
+
+In this project, I have started building a school library app. In this initial step, I implemented the classes to represent students and teachers.
+
+
+## Getting Started
```
Comment (FrederickMih) [inline README.md:8]: - [x] You'll be required to provide detailed instructions for someone without sufficient knowledge in ruby to set up your app locally.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @TuyishimireEric, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Correct Gitflow :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (TuyishimireEric) [reply README.md:8]: Thank you @FrederickMih for reviewing my project, I think I have done the required changes

**Review by Davidosky007 — APPROVED** [review]
Hi @TuyishimireEric 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All classes were added properly 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TuyishimireEric** [comment]
Thanks @Davidosky007  for approving my project

**Pull request merged**